### ISSUE TITLE
Possible index out of bound error

### ISSUE THREAD

**Issue by kaushik-rohit** [prasadtalasila/IRCLogParser#130]
In network.py "nick_receiver_from_conn_comp"
`   def nick_receiver_from_conn_comp(nick, conn_comp_list):
    for i in range(config.MAX_EXPECTED_DIFF_NICKS):
        if nick in conn_comp_list[i]:
            nick_receiver = conn_comp_list[i][0]
            break
    return nick_receiver`
On first look it seems that conn_comp_list[i] may give an index error since we don't know if config.MAX_EXPECTED_DIFF_NICKS is greater than conn_comp_list length. This should be provided with a comment about why we are using it this way or just replace it with i in range(len(conn_comp_list)).

**Comment by rohangoel96** [comment]
Yeah, this error is probably because of the way code was written in the first place by dhruv bhaiya. if the number of nick names > MAX_EXPECTED_DIFF_NICKS, there'll be an index error. Note: We can't just set it to be a very high value because that'll significantly increase the running time.

**Comment by prasadtalasila** [comment]
since we can paper results even with this bug in place, solving this bug is not a priority until paper is complete.

**Comment by prasadtalasila** [comment]
@kaushik-rohit you might be experiencing this bug during the unit tests. It is a good news if the test actually fails for this scenario. We can set a failure expectation for the test, fix the code later and then expect the test to pass.
